app/src/main/java/com/gdfp/android_evaluationv2/MainActivity.java

```diff
@@ -40,13 +41,15 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        TaskDbHelper helper =new TaskDbHelper(this);
+        mDatabase = helper.getWritableDatabase();
         setSupportActionBar(toolbar);
 
-        mAdapter = new TaskAdapter(null);
+        mAdapter = new TaskAdapter(getAllItems(),this);
         mAdapter.setOnItemClickListener(this);
 
-        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler_view);
+        RecyclerView recyclerView = findViewById(R.id.recycler_view);
```

Comment:
When you are working with legacy code like this is better to keep last implementation at least you have another good excuse or scenarios that need to update this part. It's just a recomendation.